src/services/User.service.js

```diff
@@ -9,13 +9,121 @@ class UserServices {
    * @param {string} email - email of the user
    * @returns {object} user data from database
    */
-  static async findUserByEmail(email) {
-    const user = await db.user.findOne({
-      where: { email }
-    });
+  async findUserByEmail(email) {
+    const user = await models.user.findOne({ where: { email } });
     if (!user) return null;
+    const lastLogin = new Date().toISOString();
+    await models.user.update({ lastLogin }, { where: { email } });
     return user;
   }
+
+  /**
+   * @name GetUserById
+   * @description Interacts with model to find a single user
+   * @param { string } id the user's id
+   * @returns {object} return the user's data
+   */
+  async getUserById(id) {
+    const userData = await models.user.findOne({
+      where: { id },
+      attributes: [
+        'firstName',
+        'lastName',
+        'email',
+        'isVerified',
+        'birthDate',
+        'residenceAddress',
+        'preferredLanguage',
+        'preferredCurrency',
+        'department',
+        'lineManagerId',
+        'gender',
+        'lastLogin',
+        'role',
+        'phoneNumber'
+      ]
+    });
+
+    let lineManagerData = null;
+
+    if (userData && userData.dataValues.lineManagerId) {
+      lineManagerData = await models.user.findOne({
+        where: { id: userData.dataValues.lineManagerId },
+        attributes: ['firstName', 'lastName']
+      });
+    }
+
+    return { userData, lineManagerData };
+  }
+
+  /**
+   * @name updateLastLoginByUserId
+   * @description Interacts with model to find a single user
+   * @param { object } attribute the user attribute to update
+   * @param { string } userId the user's email
+   * @returns {object} return the user's data
+   */
+  async updateLastLoginByUserId(attribute, userId) {
+    const { lastLogin } = attribute;
+    return models.profile.findOne({ where: { userId } })
+      .then(async (profile) => {
+        if (profile) {
+          await profile.update({ lastLogin });
+        }
+      });
+  }
+
+  /**
+   * @name updateUserInfoByEmail
+   * @description Interacts with model to find a single user
+   * @param { object } attribute the user attribute to update
+   * @param { string } email the user's email
+   * @param { object } res response object
+   * @returns {object} return the user's data
+   */
+  async updateUserInfoByEmail(attribute, email) {
```

Comment:
change function parameter to `attributes`.

Reply:
Let me change it that way, thanks.